runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -420,34 +412,33 @@ private ILogData cacheLoadAndGet(Cache<Long, ILogData> cache, long address, ILog
      * @return A map of addresses read, which will be cached if caching is enabled
      */
     public Map<Long, ILogData> read(Iterable<Long> addresses, @NonNull ReadOptions options) {
-        // We need this map to preserve ordering as entries are expected in ascending order.
-        Map<Long, ILogData> result = new HashMap<>();
+        if (!options.isClientCacheable()) {
+            return fetchAll(addresses, options);
```

Comment:
In the refactoring we lost the check for trimmed addresses, meaning that, in the event of cache disable we won't throw trimmed exceptions if any address was trimmed.

Reply:
Good catch! Thanks!

Reply:
Turns out that this if statement is not necessary. We already have a check in `cacheLoadAndGet`:

```
        if (!options.isClientCacheable()) {
            return loadedValue;
        }
```